source/common/filesystem/filesystem_impl.cc

```diff
@@ -171,12 +171,17 @@ void FileImpl::flushThreadFunc() {
     std::unique_lock<Thread::BasicLockable> flush_lock;
 
     {
-      std::unique_lock<std::mutex> write_lock(write_lock_);
+      Thread::LockGuard write_lock(write_lock_);
 
       // flush_event_ can be woken up either by large enough flush_buffer or by timer.
       // In case it was timer, flush_buffer_ can be empty.
       while (flush_buffer_.length() == 0 && !flush_thread_exit_) {
-        flush_event_.wait(write_lock);
+        // CondVar::wait() does not throw, and never will, as it's based on
```

Comment:
nit: I feel like we are going to end up with such a comment every time we do this. Should we move this comment to the class wait() method and then just have a comment here to look at that comment?

Reply:
done; ptal